clang/include/clang/Driver/Options.td

```diff
@@ -4063,6 +4063,11 @@ defm threadsafe_statics : BoolFOption<"threadsafe-statics",
   NegFlag<SetFalse, [], [ClangOption, CC1Option],
           "Do not emit code to make initialization of local statics thread safe">,
   PosFlag<SetTrue>>;
+defm tls_guards : BoolFOption<"tls-guards",
+  LangOpts<"TlsGuards">, DefaultTrue,
```

Comment:
If we're going to expose this as a clang option (instead of just a clang-cc1 option), this needs a better name and description.  This specifically applies to the MS ABI rules for thread-local variables not defined inside a function definition.  Maybe call it `-fms-nonlocal-tls-guards`?  Not sure.

Reply:
Ah I see :D i have never added an option before.

I removed it as a clang option and renamed it to `-fms-tls-guards` (& `-fno-ms-tls-guards`).
It applies to all tls variables in general, whether they are defined inside a function or not should not be relevant, from what I understand from Microsoft's documentation

Reply:
My understanding is that if a tls variable is defined inside a function, we don't call __dyn_tls_on_demand_init anyway.  But I don't care that much about the name.